### PR TITLE
Fix typing when inheriting from DataClassJsonMixin

### PR DESCRIPTION
**Pull request by OdinTech3**
PR fixes the below Mypy Errors: 

* `Incompatible types in assignment (expression has type "Dict[Any, Any]", base class "DataClassJsonMixin" defined the type as "None")`

* `Argument "letter_case" to "config" has incompatible type "LetterCase"; expected "Optional[Callable[[str], str]]"`

When doing something like:
```python
class MyDataClass(DataClassJsonMixin):
    dataclass_json_config = config(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)["dataclasses_json"]
```

### CODE REVIEW

**Comment by OdinTech3** [comment]
Gonna try and fix the circular import issue as soon as I can.

**Comment by USSX-Hares** [comment]
- LetterCase parameter type is reported as #286 and fixed as #343
 - `dataclass_json_config`, if I understand correctly, has no type annotation in order to not interfere the DataClasses field logic. @lidatong, am I correct?

File: dataclasses_json/cfg.py [hunk 1]
```diff
@@ -41,30 +39,32 @@ def __init__(self):
...
-        lib_metadata['encoder'] = encoder
+        lib_metadata["encoder"] = encoder
 
     if decoder is not None:
-        lib_metadata['decoder'] = decoder
+        lib_metadata["decoder"] = decoder
 
     if mm_field is not None:
-        lib_metadata['mm_field'] = mm_field
+        lib_metadata["mm_field"] = mm_field
```
Comment (USSX-Hares) [inline dataclasses_json/cfg.py:67]: Here and below: why do you change the single quotes for the double ones?

File: dataclasses_json/cfg.py [hunk 2]
```diff
@@ -41,30 +39,32 @@ def __init__(self):
...
+    # TODO: these can be typed more precisely
+    # Specifically, a Callable[A, B], where `B` is bound as a JSON type
+    encoder: Callable = None,
+    decoder: Callable = None,
+    mm_field: MarshmallowField = None,
+    letter_case: Union[Callable[[str], str], 'LetterCase'] = None,
+    undefined: Optional[Union[str, Undefined]] = None,
+    field_name: str = None,
+    exclude: Optional[Callable[[str, T], bool]] = None,
+) -> Dict[str, dict]:
```
Comment (USSX-Hares) [inline dataclasses_json/cfg.py:54]: Why do you change the existing lines indentation?

Comment (OdinTech3) [reply dataclasses_json/cfg.py:54]: I think it was because I formatted w/ black.

File: dataclasses_json/cfg.py (hunk 1 above)
Comment (OdinTech3) [reply dataclasses_json/cfg.py:67]: I think it was because I formatted w/ black.

**Comment by george-zubrienko** [comment]
I'm closing this one as stale and having unresolved comments. Feel free to open an issue and re-create the PR if you come by ;)

**Pull request closed without merging**